packages/general/src/environment/SharedEnvironmentServices.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * @license
+ * Copyright 2022-2025 Matter.js Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Environment } from "#environment/Environment.js";
+import { Environmental } from "#environment/Environmental.js";
+import { SharedServicesManager } from "#environment/SharedServicesManager.js";
+import { ImplementationError } from "#MatterError.js";
+import { MaybePromise } from "#util/index.js";
+
+/**
+ * Provides reference-counted access to shared environmental services.
+ *
+ * Tracks which services are in use and ensures services are only closed when all
+ * consumers have released them. This enables safe sharing of services across multiple
+ * consumers without premature cleanup.
+ *
+ * Created via {@link Environment.asDependent} and automatically registered at the root
+ * environment for centralized lifecycle management. All shared services are accessed
+ * at the root level regardless of which environment creates the instance.
+ */
+export class SharedEnvironmentServices {
```

Comment:
Maybe add ServiceProvider interface here to ensure that subset of API matches `Environment`

Reply:
Good idea, done